DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -1824,13 +1827,7 @@ ulong channelId
 
         RestResponse res = await this._rest.ExecuteRequestAsync(request);
 
-        DiscordChannel ret = JsonConvert.DeserializeObject<DiscordChannel>(res.Response!)!;
-
-        // this is really weird, we should consider doing this better
-        if (ret.IsThread)
-        {
-            ret = JsonConvert.DeserializeObject<DiscordThreadChannel>(res.Response!)!;
-        }
+        DiscordChannel ret = DiscordJson.ToDiscordObject<DiscordThreadChannel>(res.Response!)!;
```

Comment:
tested?

Reply:
will test when the polly issue is fixed

Reply:
tested and works flawless (tested all types)